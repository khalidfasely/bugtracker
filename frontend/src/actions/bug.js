import bugApi from "../api/bug";

export const setBug = (bug) => ({
    type: "SET_BUG",
    bug
});

export const startSetBug = (pid, bid) => {
    return (dispatch) => {
        return bugApi(pid, bid).then((result) => {
            dispatch(setBug(result.bug));
            return result;
        });
    };
};