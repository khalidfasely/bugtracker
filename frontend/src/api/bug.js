export default async (pid, bid) => {
    const res = await fetch(`/api/project/${pid}/bug/${bid}`);

    const result = await res.json();

    return result;
};