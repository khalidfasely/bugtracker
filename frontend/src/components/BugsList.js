const BugsList = ({ bug }) => (
    <div className="bug-list-item">
        <div className="bug-list-item__title-cls">
            <h3>{bug?.title}</h3>
            {
                bug?.active ?
                <p className={bug?.classification}>{bug?.classification}</p> :
                <abbr title="Not Active Anymore">
                    <p className="not-active">N.A</p>
                </abbr>
            }
        </div>
        <p className="bug-list-item__description">{bug?.description.substring(0, 120)}...</p>
        <div className="bug-list-item__user-time">
            <p>{bug?.user?.username}</p>
            <p>{bug?.time}</p>
        </div>
        {/*<div>
            {bug?.admins?.map(admin => <span key={admin}>{admin} </span>)}
        </div>
        <br />
        <div>
            {bug?.users_with?.map(user => <span key={user}>{user} </span>)}
        </div>*/}
    </div>
);

export default BugsList;