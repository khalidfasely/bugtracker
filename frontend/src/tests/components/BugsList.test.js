import React from "react";
import { render } from '@testing-library/react';

import BugsList from "../../components/BugsList";
import bug from '../fixtures/bug';

test('Should render BugsList component without bug data', () => {
    const { asFragment } = render(<BugsList />);
    expect(asFragment()).toMatchSnapshot();
});

test('Should render BugsList component with bug data', () => {
    const { asFragment } = render(<BugsList bug={bug} />);
    expect(asFragment()).toMatchSnapshot();
});