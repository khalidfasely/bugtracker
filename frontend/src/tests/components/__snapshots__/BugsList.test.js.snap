// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render BugsList component with bug data 1`] = `
<DocumentFragment>
  <div>
    <p>
      1
    </p>
    <p>
      Bug title
    </p>
    <p>
      medium
    </p>
    <p>
      now
    </p>
    <p>
      admin
    </p>
    <div>
      <span>
        admin 
      </span>
      <span>
        1 
      </span>
    </div>
    <br />
    <div>
      <span>
        admin 
      </span>
      <span>
        1 
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Should render BugsList component without bug data 1`] = `
<DocumentFragment>
  <div>
    <p />
    <p />
    <p />
    <p />
    <p />
    <div />
    <br />
    <div />
  </div>
</DocumentFragment>
`;
